### ISSUE TITLE
Installation issue: `libxsmm`

### ISSUE THREAD

**Issue by sebastiangrimberg** [LLNL/spack#41121]
### Steps to reproduce the issue

```console
$ spack spec -I libxsmm
```

### Error message

<details><summary>Error message</summary>

<pre>
==> Error: concretization failed for the following reasons:

   1. cannot satisfy a requirement for package 'libxsmm'.
</pre></details>

### Information on your system

* **Spack:** 0.22.0.dev0 (1927ca1f3504468f072573e90db0f12c58306098)
* **Python:** 3.11.5
* **Platform:** darwin-ventura-m1
* **Concretizer:** clingo

### Additional information

PR https://github.com/spack/spack/pull/40494 introduced the following into `libxsmm/package.py`:

```
# Intel Architecture or compatible CPU required
requires("target=x86_64:")
```

However, I do not think this is required; I regularly build and use LIBXSMM on Mac M1 and other Arm systems. I believe LIBXSMM supports a wide range of systems beyond just x64_64 (https://libxsmm.readthedocs.io/en/latest/libxsmm_compat/). Can this constraint be removed in the Spack recipe?

Tagging @hfp as the package maintainer and @haampie as the author of #40494.

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

**Comment by haampie** [comment]
Would you be willing to add support for it? It errored on build before #40494

**Comment by sebastiangrimberg** [comment]
What platforms did you observe build errors for? I am able to build `libxsmm` with Spack once I revert the changes introduced by #40494 on non-x86_64 platforms just fine (on AWS Graviton as well as Mac M1). Is there perhaps a more refined conflict suggestion you could use to replace `requires("target=x86_64:")` in the `libxsmm` recipe? I guess my point is that this is overly restrictive as `libxsmm` supports non-x86_64 systems as far as I can tell.

**Comment by RMeli** [comment]
Should be fixed by #41193. Thanks for reporting @sebastiangrimberg.
